src/XAxis.re

```diff
@@ -8,49 +8,155 @@ external make:
     ~allowDataOverflow: bool=?,
     ~allowDecimals: bool=?,
     ~allowDuplicatedCategory: bool=?,
-    ~axisLine: 'axisLine=?,
+    ~axisLine:
+      [@mel.unwrap] [
+        | `Bool(bool)
+        | `Obj(Js.t({..}))
+      ]
+        =?,
     ~className: string=?,
-    ~dataKey: string=?,
+    ~dataKey:
+      [@mel.unwrap] [
+        | `Str(string)
+        | `Int(int)
+        | `Fn('dataObj => 'data)
+      ]
+        =?,
     ~domain: array('domain)=?,
     ~height: int=?,
     ~hide: bool=?,
     ~interval: AxisInterval.t=?,
-    ~label: 'label=?,
+    ~label:
+      [@mel.unwrap] [
```

Comment:
Should we maybe replace some of those types in `Utils` with `mel.unwrap` too?

https://github.com/ahrefs/melange-recharts/blob/master/src/Utils.re#L135

Reply:
I'd like to tackle this in another PR but I think this is a great suggestion.

Reply:
PR open: https://github.com/ahrefs/melange-recharts/pull/60